Compiling with recent Clang reports missing-template-arg-list-after-template-kw

MacOS is updating its clang version in recent updates, causing the following compile error to emerge

```
.../yalantinglibs/include/iguana/detail/itoa.hpp:222:33: error: a template argument list is expected after a name prefixed by the template keyword [-Wmissing-template-arg-list-after-template-kw]
  222 |     return convert<D>::template itoa(p, u);
      |                                 ^
.../yalantinglibs/include/iguana/detail/itoa.hpp:260:30: error: a template argument list is expected after a name prefixed by the template keyword [-Wmissing-template-arg-list-after-template-kw]
  260 |     p = convert<D>::template itoa(p, u);
      |                              ^
```

It seems like others have encountered this issue (https://github.com/llvm/llvm-project/issues/94194 https://github.com/llvm/llvm-project/issues/101079) and LLVM insists that this behavior is correct.

--- a/include/ylt/standalone/iguana/detail/itoa.hpp
+++ b/include/ylt/standalone/iguana/detail/itoa.hpp
@@ -219,7 +219,7 @@ struct convert {
   template <typename U, std::enable_if_t<!std::is_signed<U>::value &&
                                          std::is_integral<U>::value>* = nullptr>
   static inline char* itoa(U u, char* p) {
-    return convert<D>::template itoa(p, u);
+    return convert<D>::template itoa<>(p, u);
   }
 
   // itoa: handle signed integral operands (selected by SFINAE)
@@ -257,7 +257,7 @@ struct convert {
       *p = '-';
       p += (mask & 1);
     }
-    p = convert<D>::template itoa(p, u);
+    p = convert<D>::template itoa<>(p, u);
     if (D == Rev && mask)
       *--p = '-';
     return p;

